src/main/java/com/algosenpai/app/UserStorageParser.java

```diff
@@ -0,0 +1,66 @@
+package com.algosenpai.app;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.util.Scanner;
+
+public class UserStorageParser {
+
+    //The scanner which is used for reading in from the file.
+    private Scanner inputScanner;
+
+    /**
+     * Initialises everything so that Scanner works.
+     * @throws FileNotFoundException will never be thrown as the file would already be prepared.
+     */
+    UserStorageParser() throws FileNotFoundException {
+        String workingDir = System.getProperty("user.dir") + File.separator + "UserData.txt";
+        FileReader reader = new FileReader(workingDir);
+        this.inputScanner = new Scanner(reader);
+    }
+
+    /**
+     * Responsible for parsing in the input for each chapter. The skips are used to remove unnecessary information
+     * from each line of input. All the respective information needed for each ChapterStat is derived and constructed.
+     *
+     * @return the ChapterStat for each chapter.
+     */
+    public ChapterStat nextChapterStat() {
```

Comment:
First sentence of javadoc comments should be a short summary of what's the end-result of executing this method, and should start with a word that's singular verb.

You can read up on this particular section of the coding standard [here](https://nuscs2113-ay1920s1.github.io/website/coding-standards/java/intermediate.html#comments).